From a663e35e688bda4a6841271cbdfbb287654aecd4 Mon Sep 17 00:00:00 2001
From: jules <jules@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Thu, 5 May 2011 15:56:06 +0000
Subject: [PATCH] 	gcc/
 	* config/arm/neon.md (vec_set<mode>_internal): Fix misplaced
 	parenthesis in D-register case.

	gcc/testsuite/
	* gcc.target/arm/neon-vset_lanes8.c: New test.



git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_6-branch@173439 138bc75d-0d04-0410-961f-82ee72b054a4

index bc13722..491dd98 100644
--- a/gcc/config/arm/neon.md
+++ b/gcc/config/arm/neon.md
@@ -425,7 +425,7 @@
           (match_operand:SI 2 "immediate_operand" "i")))]
   "TARGET_NEON"
 {
-  int elt = ffs ((int) INTVAL (operands[2]) - 1);
+  int elt = ffs ((int) INTVAL (operands[2])) - 1;
   if (BYTES_BIG_ENDIAN)
     elt = GET_MODE_NUNITS (<MODE>mode) - 1 - elt;
   operands[2] = GEN_INT (elt);
new file mode 100644
index 0000000..e87102e
--- /dev/null
+++ b/gcc/testsuite/gcc.target/arm/neon-vset_lanes8.c
@@ -0,0 +1,21 @@
+/* Test the `vset_lane_s8' ARM Neon intrinsic.  */
+
+/* { dg-do run } */
+/* { dg-require-effective-target arm_neon_hw } */
+/* { dg-options "-O0" } */
+/* { dg-add-options arm_neon } */
+
+#include "arm_neon.h"
+#include <stdlib.h>
+#include <string.h>
+
+int8x8_t x = { 1, 2, 3, 4, 5, 6, 7, 8 };
+int8x8_t y = { 1, 2, 3, 16, 5, 6, 7, 8 };
+
+int main (void)
+{
+  x = vset_lane_s8 (16, x, 3);
+  if (memcmp (&x, &y, sizeof (x)) != 0)
+    abort();
+  return 0;
+}
-- 
1.7.0.4

